openkfw/TruBudget
# 1863 upgrade button

### Checklist




### Description


### How to test

1. Start app using upgradable development script 
2. Create file upgradadble.txt in /api/trubudget-config
3. Login as root and go to service status and click th upgrade button
4. check if file is created in /api/trubudget-config folder
5. Now we validated creation of file
6. Now stop app and run it again via operation script (set older version e.g. v2.12.0 in env)
7. Start app by start-upgradable-trubudget.sh script
8. Now manually create the same file `upgrade_version.txt` with content "v2.14.0"  (if the file was deleted in the meantime)
9. Wait about 3 minutes
10. App should restart with new version, v2.14.0
11. File should be deleted
12. Check if app is not restarting more then ones

Alternatively if app is started with `start-upgradable-trubudget.sh` script it has the ability to restart itself when stopped. The check is based on presence of running api container. If you stop it, the app should restart.
You can stop this process by `stop-upgradable-trubudget.sh` script.




Closes #1863
